pyomo/util/latex_printer.py

```diff
@@ -0,0 +1,1484 @@
+            )
+        ]
+        constraints = [
+            con
+            for con in pyomo_component.component_objects(
+                pyo.Constraint, descend_into=True, active=True
+            )
+        ]
+        expressions = []
+        templatize_fcn = templatize_constraint
+
+    else:
+        raise ValueError(
+            "Invalid type %s passed into the latex printer"
+            % (str(type(pyomo_component)))
+        )
+
+    if isSingle:
+        temp_comp, temp_indexes = templatize_fcn(pyomo_component)
+        variableList = []
+        for v in identify_components(
+            temp_comp, [ScalarVar, _GeneralVarData, IndexedVar]
+        ):
+            if isinstance(v, _GeneralVarData):
+                v_write = v.parent_component()
+                if v_write not in ComponentSet(variableList):
+                    variableList.append(v_write)
+            else:
+                if v not in ComponentSet(variableList):
+                    variableList.append(v)
+
+        parameterList = []
+        for p in identify_components(
+            temp_comp, [ScalarParam, _ParamData, IndexedParam]
+        ):
+            if isinstance(p, _ParamData):
+                p_write = p.parent_component()
+                if p_write not in ComponentSet(parameterList):
+                    parameterList.append(p_write)
+            else:
+                if p not in ComponentSet(parameterList):
+                    parameterList.append(p)
+
+        # Will grab the sets as the expression is walked
+        setList = []
+
+    else:
+        variableList = [
+            vr
+            for vr in pyomo_component.component_objects(
+                pyo.Var, descend_into=True, active=True
+            )
+        ]
+
+        parameterList = [
+            pm
+            for pm in pyomo_component.component_objects(
+                pyo.Param, descend_into=True, active=True
+            )
+        ]
+
+        setList = [
+            st
+            for st in pyomo_component.component_objects(
+                pyo.Set, descend_into=True, active=True
+            )
+        ]
+
+    descriptorDict = {}
+    if use_short_descriptors:
+        descriptorDict['minimize'] = '\\min'
+        descriptorDict['maximize'] = '\\max'
+        descriptorDict['subject to'] = '\\text{s.t.}'
+        descriptorDict['with bounds'] = '\\text{w.b.}'
+    else:
+        descriptorDict['minimize'] = '\\text{minimize}'
+        descriptorDict['maximize'] = '\\text{maximize}'
+        descriptorDict['subject to'] = '\\text{subject to}'
+        descriptorDict['with bounds'] = '\\text{with bounds}'
+
+    # In the case where just a single expression is passed, add this to the constraint list for printing
+    constraints = constraints + expressions
+
+    # Declare a visitor/walker
+    visitor = _LatexVisitor()
+
+    variableMap = ComponentMap()
+    vrIdx = 0
+    for i in range(0, len(variableList)):
+        vr = variableList[i]
```

Comment:
It doesn't look like you use `i`? So I think you just need `for vr in variableList:`

Reply:
Will check all of these  and the enumerates.  There is an issue here with python defaulting to a copy of array elements that I have to be careful of